app/script/view_model/ActionsViewModel.js

```diff
@@ -24,29 +24,84 @@ window.z.viewModel = z.viewModel || {};
 
 z.viewModel.ActionsViewModel = class ActionsViewModel {
   constructor(mainViewModel, repositories) {
-    this.clientRepository = repositories.client;
-    this.conversationRepository = repositories.conversation;
-    this.integrationRepository = repositories.integration;
-    this.userRepository = repositories.user;
+    const {
+      client: clientRepository,
+      conversation: conversationRepository,
+      integration: integrationRepository,
+      user: userRepository,
+    } = repositories;
+
+    this.clientRelated = {
+      deleteClient: clientRepository.deleteClient,
+    };
+
+    const {
+      archiveConversation,
+      clear_conversation,
+      deleteMessage,
+      deleteMessageForEveryone,
+      get1To1Conversation,
+      removeMember,
+      setNotificationState,
+      unarchiveConversation,
+      updateParticipatingUserEntities,
+    } = conversationRepository;
+    this.conversationRelated = {
+      archiveConversation,
```

Comment:
I am not sure the destructuring/restructuring is a good idea. It will easily lead to `this` being bound to the wrong object (it will be bound to the arbitrary object, not to the repo anymore)